Update Rails, Ruby, and add ruby debug gem

ahangarha opened the pull request:
In this PR:
- [x] Updated Rails to version 7+
- [x] Fixed builder missing template error
- [x] Added debug gem

Possible todos:
- [x] Update node version to 18.13.0 (LTS)
- [x] ~Update Ruby version to the latest: 3.2~
    Our CI still doesn't support Ruby 3.2.0
- [x] Update docker files
- [ ] Update README
- [ ] Update docs (jbuilder)
- [ ] Update changelog?
Closes #515 

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/516)
<!-- Reviewable:end -->

justin808 commented:
Did you follow the steps at https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-6-1-to-rails-7-0?

ahangarha commented:
@justin808 
I followed the mentioned upgrade guideline. Just there are two concerns:
- Updating [cache serialization format](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#new-activesupport-cache-serialization-format) doesn't work. I couldn't make the test pass for this change so I omitted this change.
- There is new change in [test environment config](https://github.com/shakacode/react-webpack-rails-tutorial/blob/515-update-rails-7-ruby-31-and-add-ruby-debug-gem/config/environments/test.rb#L19) file related to eager loading. Since I was not sure about the right value, I kept the old value to work and added the new code as a comment.
- I noticed an inconsistency in our CI tests. The same changes could sometimes pass and sometimes fail. Namely, the commit [d825126](https://github.com/shakacode/react-webpack-rails-tutorial/pull/516/commits/d82512673cef6d9363c4fa9c0f7821debbd67036) here (which passes the CI) failed twice at first. I removed the commit and made it again and it it failed. I reran the CI and it passed. I am not sure what causes this issue but as per the log, it could be related to rspec-rails version.

ahangarha commented:
Few more concerns:
- [ ] Our Jbuilder doc file needs to be updated. I edited the to be the same way used in pages_controller. I added a new react component in erb one of our views but it didn't work. (though I generated the correct data but the component failed in getting rendered on the page. This needs further investigation.
- [ ] What to do with Travis config? It is very outdated and I think we can remove it.
- [ ] It seems we don't use Codeship anymore. If so, let's remove the badge on README file and also its outdated documentation.
- [ ] It seems our Gemnasium badge is broken.
